### ISSUE TITLE
Add scraper for Santa Clara County, California

### ISSUE THREAD

**Issue by 1ec5** [covidatlas/coronadatascraper#965]
## Location name
Santa Clara County, CA, USA

## Source URL
https://www.sccgov.org/sites/covid19/Pages/dashboard.aspx

## Notes/comments
There used to be a scraper for this county, but it was [deprecated and discontinued](https://github.com/covidatlas/coronadatascraper/blob/e5dd5c7b8a5b551dd588562995c12c85e0b6740c/src/shared/scrapers/US/CA/santa-clara-county.js#L28-L31) in favor of [the _Mercury News_ spreadsheet](https://github.com/covidatlas/coronadatascraper/blob/e5dd5c7b8a5b551dd588562995c12c85e0b6740c/src/shared/scrapers/US/CA/_mercury-news.js#L45-L46), which aggregates all the counties’ reports.

The original scraper was discontinued because the county switched to a Power BI dashboard that’s difficult to scrape off a webpage. Over on Wikimedia Commons, [this table](https://commons.wikimedia.org/wiki/Data:COVID-19_Cases_in_Santa_Clara_County,_California.tab) is updated semi-automatically using [this script](https://gist.github.com/1ec5/291a1594d27fa783eaa283a2e529ab75) that was derived from network requests the dashboard makes. A scraper here could take the same approach, but it would be a little easier to read as JavaScript instead of `jq` invocations.

As detailed in sfbrigade/stop-covid19-sfbayarea#92, the _Mercury News_ spreadsheet reports changes to the cumulative case count over time, which differs from the county’s official methodology that retroactively attributes cases to past days. When an application visualizes the time series in a bar graph, the depicted curve can lead to different conclusions than the official curve.

At any given time, there are a couple dozen undated cases, which are not accounted for in the time series (since it would always result in an artificial spike at the end). The current case count is equal to yesterday’s case count plus the new cases attributed to today (not shown on the dashboard, not the same as the “New Cases” at the top of the dashboard) plus the undated case count.

In other words, the last entry in the time series always falls a couple dozen short of the current case count, making it inconsistent with sources like the _Mercury News_ that are focused on reporting the current case count accurately. Meanwhile, the county dashboard only includes a time series for confirmed case counts and test counts, but not for deaths or hospitalizations. Would these inconsistencies prevent the county dashboard from being used as a source for this project?

/cc @jbencina @HarrietRowan

**Comment by jzohrab** [comment]
Hi @1ec5 , thank you, great issue detail!

The mercury news scraper appears to scrape obsolete data ... we need to scrape the Timeseries tab in that spreadsheet.

I don't think we'd want to rely on wikimedia, but the other calls look great.  PowerBI is tough. :-)  Do you think we could fill in a good dataset using just PowerBI?

Cheers and regards, jz

**Comment by 1ec5** [comment]
> I don't think we'd want to rely on wikimedia, but the other calls look great. PowerBI is tough. :-) Do you think we could fill in a good dataset using just PowerBI?

Yes, [this script](https://gist.github.com/1ec5/291a1594d27fa783eaa283a2e529ab75) is designed to update the Wikimedia Commons table, keeping that table’s structure, but it actually fetches enough information to populate a new table on its own. It’s also possible to extract data about testing, though I haven’t gotten around to it because the Commons table didn’t require it.

**Comment by jzohrab** [comment]
Thank you for the info, this is very helpful!  We're swamped of course, but will get to this when we can.  Too many priorities.  If you know of any js coders who can contribute, we gladly take contributions.  :-)  Cheers! z
